antsibull/config.py

```diff
@@ -142,17 +141,23 @@ def convert_to_none(cls, value):
         return value
 
 
-def find_config_files(conf_files: t.Iterable[str]) -> t.List[str]:
+def find_config_files(conf_files: t.Iterable[str],
+                      user_supplied_conf_files: t.Optional[t.Iterable[str]] = None) -> t.List[str]:
```

Comment:
I think adding this to the parameters is leaking implementation details into the api.  Probably better to just remove the expanduser and expandvars calls from here and add expanduser to line 30, where the hardcided definition is made.

Reply:
I almost implemented that, except that I moved the `expanduser` call to inside `load_config` and not to line 30/31. The reasoning is that it could happen that a caller wants to modify `os.environ` (and add/adjust `HOME` in it) before calling `load_config`.